src/GitHubUtil.js

```diff
@@ -13,16 +13,50 @@ const logApiResult = Log.logApiResult;
 
 const GitHubAuthentication = { type: 'token', username: Config.githubUserLogin(), token: Config.githubToken() };
 
+function commonAppender(allPages, aPage) {
+    allPages.data = allPages.data.concat(aPage.data);
+}
+
+function statusAppender(statusPages, aPage) {
+    let statuses = statusPages.data.statuses;
+    statusPages.data.statuses = statuses.concat(aPage.data.statuses);
+}
+
+function protectedBranchAppender(protectedBranchPages, aPage) {
+    let contexts = protectedBranchPages.data.protection.required_status_checks.contexts;
+    protectedBranchPages.data.protection.required_status_checks.contexts =
```

Comment:
Why do we need these custom appending functions? Are we doing something wrong elsewhere that makes commonAppender inapplicable to branches and status checks? The official Pagination [example](https://www.npmjs.com/package/@octokit/rest#pagination) does not mention the need for custom appenders but perhaps it is oversimplified.

Reply:
This is needed because the returned by an API call data layout may be different. In the example, each res.data is an array which is concatenated, but statuses and protected branch APIs have paginated arrays somewhere deep inside res.data. Only the caller (i.e., an API context) know what array is going to be concatenated.